microsoft_auth/client.py

```diff
@@ -172,13 +169,54 @@ def authorization_url(self):
 
         return super().authorization_url(auth_url, response_mode="form_post")
 
+    def get_exp_datetime(self):
+        exp = datetime.datetime.utcnow() + datetime.timedelta(seconds=300)
+        return exp
+
+    def create_assertion(self):
+        """Creates JWT assertion"""
+
+        exp_at = self.get_exp_datetime()
+
+        payload = {
+            "sub": self.config.MICROSOFT_AUTH_CLIENT_ID,
+            "iss": self.config.MICROSOFT_AUTH_CLIENT_ID,
+            "aud": self.openid_config["token_endpoint"],
+            "exp": exp_at,
+        }
+
+        thumbprint = self.config.MICROSOFT_AUTH_ASSERTION_CERTIFICATE_THUMBPRINT
+        key = self.config.MICROSOFT_AUTH_ASSERTION_KEY_CONTENT
+
+        assertion = jwt.encode(
+            payload, key, algorithm="RS256", headers={"x5t": thumbprint}
+        )
+
+        return assertion
+
     def fetch_token(self, **kwargs):
         """Fetchs OAuth2 Token with given kwargs"""
 
+        if self.config.MICROSOFT_AUTH_ASSERTION_CERTIFICATE != "":
+            assertion = self.create_assertion()
+
+            client_type = (
+                "urn:ietf:params:oauth:client-assertion-type:jwt-bearer"  # noqa
+            )
+
+            arg = {
+                "cert": self.config.MICROSOFT_AUTH_ASSERTION_CERTIFICATE,
+                "include_client_id": True,
+                "client_id": self.config.MICROSOFT_AUTH_CLIENT_ID,
+                "client_assertion_type": client_type,
+                "client_assertion": assertion,
+            }
```

Comment:
You should add these args into `kwargs` to prevent duplicate kwargs to `fetch_token`

(sorry, reposted to get rid of "review")